test/Microsoft.ML.Tests/SvmLightTests.cs

```diff
@@ -495,7 +497,7 @@ public void TestSvmLightLoaderLongIndex()
                 new SvmLightOutput() { Label = -1, Weight = 1, Features = new VBuffer<float>(6, 1, new[] { 5f }, new[] { 1 }) }
             }, schemaDef);
             var savingPath = DeleteOutputPath(TestName + "-saved-data.txt");
-            TestSvmLight(path, savingPath, 0, int.MaxValue, false, expectedData);
+            TestSvmLight(path, savingPath, 0, 6, false, expectedData);
```

Comment:
Earlier you had explained to me that the right fix was to change the size of the VBuffer to be int.MaxValue. Can you please explain why this line had to change? Does this implies that there is now no test for key with int.MaxValue or we don't support keys with int.MaxValue?

Reply:
We support the same scenarios as we did before, except now without a bug :-).

The old behavior was that if while loading we encounter an index greater than `int.MaxValue`, then we define the length of the Features column to be `int.MaxValue`. The change this PR is making is to define the length of Features column to be the maximum encountered index that is at most `int.MaxValue`, because the old behavior is what caused the issue that when you save and reload, the length of the Features column might change.


---
In reply to: [369680210](https://github.com/dotnet/machinelearning/pull/4690#discussion_r369680210) [](ancestors = 369680210)